src/ui/run-history-ui-handler.ts

```diff
@@ -0,0 +1,338 @@
+    super.showText(text, delay, callback, callbackDelay, prompt, promptDelay);
+
+    if (text?.indexOf("\n") === -1) {
+      this.runSelectMessageBox.setSize(318, 28);
+      this.message.setY(-22);
+    } else {
+      this.runSelectMessageBox.setSize(318, 42);
+      this.message.setY(-37);
+    }
+
+    this.runSelectMessageBoxContainer.setVisible(!!text?.length);
+  }
+
+  setCursor(cursor: integer): boolean {
+    const changed = super.setCursor(cursor);
+
+    if (!this.cursorObj) {
+      this.cursorObj = this.scene.add.nineslice(0, 0, "select_cursor_highlight_thick", undefined, 296, 44, 6, 6, 6, 6);
+      this.cursorObj.setOrigin(0, 0);
+      this.runsContainer.add(this.cursorObj);
+    }
+    this.cursorObj.setPosition(4, 4 + (cursor + this.scrollCursor) * 56);
+
+    return changed;
+  }
+
+  setScrollCursor(scrollCursor: integer): boolean {
+    const changed = scrollCursor !== this.scrollCursor;
+
+    if (changed) {
+      this.scrollCursor = scrollCursor;
+      this.setCursor(this.cursor);
+      this.scene.tweens.add({
+        targets: this.runsContainer,
+        y: this.runContainerInitialY - 56 * scrollCursor,
+        duration: Utils.fixedInt(325),
+        ease: "Sine.easeInOut"
+      });
+    }
+
+    return changed;
+  }
+
+  clear() {
+    super.clear();
+    this.runSelectContainer.setVisible(false);
+    this.eraseCursor();
+    this.runSelectCallback = null;
+    this.clearruns();
+  }
+
+  eraseCursor() {
+    if (this.cursorObj) {
+      this.cursorObj.destroy();
+    }
+    this.cursorObj = null;
+  }
+
+  clearruns() {
+    this.runs.splice(0, this.runs.length);
+    this.runsContainer.removeAll(true);
+  }
+}
+
+class RunEntry extends Phaser.GameObjects.Container {
+  public slotId: integer;
+  public hasData: boolean;
+  public entryData: RunHistoryData;
+  private loadingLabel: Phaser.GameObjects.Text;
+
+  constructor(scene: BattleScene, runHistory: any, timestamp: string, slotId: integer) {
+    super(scene, 0, slotId*56);
+
+    this.slotId = slotId;
+    this.hasData = true;
+    this.entryData = runHistory[timestamp];
+
+    this.setup(this.entryData);
+
+  }
+
+  setup(run: any) {
+
+    const victory = run.victory;
+    const data = this.scene.gameData.parseSessionData(JSON.stringify(run.entry));
+
+    const slotWindow = addWindow(this.scene, 0, 0, 304, 52);
+    this.add(slotWindow);
+
+
+    if (victory) {
+      const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:victory")}`, TextStyle.WINDOW);
+      this.add(gameOutcomeLabel);
+    } else {
+      if (data.battleType === BattleType.WILD) {
+        const enemyContainer = this.scene.add.container(8, 5);
+        const gameOutcomeLabel = addTextObject(this.scene, 0, 0, `${i18next.t("runHistory:defeatedWild")}`, TextStyle.WINDOW);
+        enemyContainer.add(gameOutcomeLabel);
+        data.enemyParty.forEach((enemyData, e) => {
+          //This allows the enemyParty to be shown - doubles or sings -> 58+(e*8)
```

Comment:
```suggestion
          // This allows the enemyParty to be shown - doubles or singles -> 58+(e*8)
```
Presumably you meant "singles" and not "sings"? 🤔